pkg/util/util.go

```diff
@@ -323,3 +323,19 @@ func GenerateUniqueHashForWorkloadImageTag(namespace string) string {
 	h.Write([]byte(namespace))
 	return base64.URLEncoding.EncodeToString(h.Sum(nil))[0:5]
 }
+
+// GenerateUniqueRouteName returns a unique, trimmed route name based on the Component name based on the following criteria
+// 1. Under 30 characters
+// 2. Contains 4 random characters
+func GenerateUniqueRouteName(componentName string) string {
```

Comment:
How do we ensure this is unique across all environments? do you mean stage and prod of unique across project namespace?

Reply:
Random instead of unique perhaps?

Reply:
yes, `GenerateRandomRouteName` may be better.  I was just trying to clarify the scope of this fix according to your description.  I wasn't sure if the intention was to check across all environments for uniqueness

> 3. Are unique across all environments for a given Component

Reply:
Yeah, I’ll clarify that. Given that we can poll the namespace on other clusters for route names, this should be sufficient for now

Reply:
Updated.